OSDOCS-2457: document procedures for Global Options to Enable HTTP Strict Transport Security (HSTS) (#35812)
openshift/openshift-docs: modules/nw-enabling-hsts-global.adoc

// Module included in the following assemblies:
// * networking/configuring-routing.adoc

[id="nw-enabling-hsts-global_{context}"]

> = Enabling HTTP strict transport security globally

Reviewer: ```suggestion
= Enforcing HTTP strict transport security globally
```
It's more accurate to say "enforcing" because it does not actually change anything automatically.  It just adds a policy that will let the user know if they have tried to configure a route that doesn't follow the HSTS policy.

> To enable HTTP strict transport security (HSTS) globally for all TSL routes, add a new type `IngressControllerHSTSHeaderPolicy` to the API in `openshift/api/blob/master/operator/v1/types_ingress.go` to capture the configuration of the HSTS Policy.

Reviewer: ```suggestion
To enable HTTP strict transport security (HSTS) globally for all TLS routes, add a `requiredHSTSPolicies` record to the Ingress spec to capture the configuration of the HSTS Policy.
```

> If you have several routes or regulatory compliance requirements, you can use the global method for configuring HSTS.

Reviewer: ```suggestion
If you have several routes or regulatory compliance requirements, you can use the global method for configuring HSTS.
```
This is not true.

Author: Removed, thanks!

.Prerequisites

* You are logged in to the cluster with a user with `cluster-admin` privileges.

.Procedure

> . Edit the Ingress conifg file:

Reviewer: ```suggestion
. Edit the Ingress config file:
```

+
[source,terminal]
----
$ oc edit ingresses.config.openshift.io/cluster
----
+

> . To enable HSTS globally for all TLS routes, set the `scope` field to `all`:

Reviewer: This is obsolete information from the old/replaced enhancement proposal.  The new proposal is at https://github.com/openshift/enhancements/pull/849/files (with some updates that are still not merged).
